### DISCUSSION TITLE
搜索不出来结果

### DISCUSSION THREAD

**Discussion by zouhuigang** [milvus-io/milvus#30026] [Q&A and General discussion]
<img width="1305" alt="image" src="https://github.com/milvus-io/milvus/assets/15059874/cc03342c-b61b-49b5-a5be-f5ac977865a6">

向量搜索”手动排样“搜索不出来结果，但是手动查询是有的
<img width="1220" alt="image" src="https://github.com/milvus-io/milvus/assets/15059874/89dd24f6-b181-4b10-bda2-2997df42de3c">

embedding：[0.0075235809199512005,0.020097026601433754,-0.024473577737808228,0.007609376683831215,-0.03074987418949604,0.014702336862683296,0.018310798332095146,-0.012197809293866158,-0.014147668145596981,-0.0381350964307785,-0.011350099928677082,-0.03251826763153076,0.003349058097228408,-0.0046163261868059635,-0.028427626937627792,0.0012065267655998468,0.0020569998305290937,-0.05288185179233551,0.014148721471428871,-0.000821401656139642,-0.0533217191696167,-0.0014411279698833823,-0.022398700937628746,-0.007463714573532343,0.023631911724805832,0.04845384880900383,-0.014638153836131096,0.00890979915857315,0.009395231492817402,-0.019347960129380226,0.04046918451786041,0.011347045190632343,0.002459268318489194,0.007040193770080805,0.030001001432538033,0.04486008733510971,0.017045674845576286,0.04453706368803978,0.05702241137623787,0.021271850913763046,0.02569487877190113,0.0011354904854670167,0.01589040271937847,-0.011018377728760242,-0.01704595796763897,-0.00784278754144907,-0.0012354792561382055,0.011698756366968155,-0.039470501244068146,-0.02037901058793068,-0.0001315123081440106,0.058704573661088943,-0.03563855215907097,-0.058151502162218094,-0.01334155723452568,0.010794470086693764,-0.03262101113796234,-0.042347490787506104,-0.04323136806488037,0.10310289263725281,0.0008541282149963081,-0.01593700796365738,-0.042989540845155716,-0.0939648300409317,0.05049753934144974,0.023921553045511246,-0.03464322164654732,0.049341119825839996,-0.027110205963253975,0.03060159832239151,0.0143608832731843,-0.04322149232029915,-0.05399230495095253,0.007459097541868687,0.019892368465662003,-0.009194298647344112,-0.00889493990689516,0.008560353890061378,-0.0486171655356884,0.020016135647892952,0.014584241434931755,0.041681718081235886,-0.039098143577575684,-0.020076988264918327,0.028033392503857613,0.0029277061112225056,0.030128151178359985,-0.038351863622665405,0.03586921468377113,0.05561913922429085,-0.0028175527695566416,0.022581107914447784,-0.08664607256650925,-0.008317881263792515,0.04196199029684067,0.04702425375580788,-0.017345594242215157,0.012630543671548367,-0.014742729254066944,0.02569526806473732,0.030315963551402092,0.004070280119776726,-0.0025721683632582426,-0.013061347417533398,0.026218988001346588,-0.005803306587040424,0.015873292461037636,-0.03579355776309967,-0.014938397333025932,0.017596954479813576,0.013367732986807823,-0.023967113345861435,-0.016870401799678802,0.02148798480629921,0.010817909613251686,0.02658906579017639,0.0356411375105381,-0.01137083861976862,-0.045011188834905624,0.07065866887569427,-0.0024307300336658955,0.07138387113809586,-0.04435750097036362,-0.0077962568029761314,-0.023468703031539917,0.03090439736843109,-0.05073392391204834,0.0035928450524806976,-0.011367238126695156,-0.06585900485515594,0.013044227845966816,-0.00013073766604065895,0.051725853234529495,-0.004418640397489071,-0.03840120881795883,-0.008097496815025806,-0.004788332153111696,-0.0254234466701746,-0.02608361840248108,-0.02570977993309498,-0.0034957185853272676,0.0008285066578537226,0.013803220354020596,0.004321490414440632,-0.012163043022155762,0.012777332216501236,-0.02340484969317913,-0.006942236330360174,0.024170586839318275,0.029408879578113556,0.0035889004357159138,0.07651421427726746,-0.03700133413076401,-0.03344811871647835,0.012328566052019596,0.01514059491455555,-0.20413391292095184,-0.008739990182220936,0.006584314629435539,-0.0475945845246315,0.021027015522122383,-0.07598113268613815,0.008344210684299469,0.01941218413412571,0.07698293030261993,0.035049326717853546,0.03228963911533356,0.03915313631296158,0.0065572746098041534,0.004423959646373987,0.016275819391012192,0.01465484220534563,0.013263225555419922,-0.013289021328091621,0.04711156710982323,-0.04833181947469711,-0.037801746279001236,0.056181054562330246,0.015467549674212933,-0.054035212844610214,-0.011368311941623688,0.03300194442272186,0.0007231775671243668,-0.0033738254569470882,0.010866383090615273,-0.006402072496712208,0.016048908233642578,0.029167992994189262,-0.022190306335687637,0.012741231359541416,-0.002544121351093054,-0.0011561161372810602,-0.011863227002322674,-0.010295483283698559,-0.0056372289545834064,0.06467645615339279,-0.013911951333284378,0.009453032165765762,0.004519195295870304,-0.00016360908921342343,-0.03506266325712204,-0.027839910238981247,0.015092838555574417,0.006884655449539423,-0.014910431578755379,0.08851262181997299,-0.015514733269810677,-0.012713832780718803,-0.006015865132212639,0.025558138266205788,0.015011375769972801,-0.03111090138554573,-0.0041618370451033115,0.009760447777807713,0.03378431126475334,-0.019021349027752876,0.02836548537015915,0.006741293240338564,-0.03140751272439957,-0.006023511290550232,0.032097525894641876,-0.01907951943576336,0.0537547692656517,0.018615761771798134,0.0150371752679348,-0.028086356818675995,0.048652634024620056,0.028206631541252136,0.02240932546555996,-0.01494924072176218,-0.025679316371679306,0.010778469033539295,0.018650129437446594,-0.04519636183977127,0.015372802503407001,0.04068281129002571,-0.0345848947763443,0.0018180679762735963,0.011011693626642227,0.025061756372451782,0.06603387743234634,0.008825994096696377,0.02492961287498474,-0.007305916864424944,-0.02749018929898739,0.009651575237512589,-0.00399796525016427,-0.002978226402774453,-0.01775420643389225,0.02768312767148018,-0.016583142802119255,-0.019996529445052147,0.028524916619062424,0.0049756900407373905,0.02222740650177002,0.06087895855307579,0.004785266704857349,-0.023082753643393517,0.007545699365437031,0.025181008502840996,0.02045332081615925,0.008527441881597042,-0.024585235863924026,-0.08267814666032791,0.020816344767808914,-0.022082922980189323,-0.056267235428094864,0.024140184745192528,-0.016334520652890205,0.03787797689437866,-0.052269693464040756,0.01321815513074398,-0.04460854455828667,0.01249778177589178,0.010764396749436855,0.026666460558772087,-0.01842789724469185,0.0018617247696965933,-0.02370084449648857,0.08667222410440445,0.022319182753562927,-0.020924421027302742,0.01647135615348816,0.036264438182115555,-0.02440343238413334,-0.023814015090465546,0.024922922253608704,0.03181423619389534,0.047340113669633865,0.042038291692733765,-0.013122533448040485,0.04015818238258362,0.03732693940401077,-0.018892504274845123,-0.00954409409314394,-0.016872864216566086,0.0313301756978035,-0.02581225335597992,-0.017851734533905983,0.012755162082612514,-0.00401701033115387,0.041579317301511765,-0.020815927535295486,-0.01883731409907341,0.06103881448507309,0.013307693414390087,0.07313931733369827,-0.03405674174427986,-0.016264017671346664,-0.0268302820622921,0.012705892324447632,-0.024192068725824356,0.0013990094885230064,0.06904202699661255,-0.002540294546633959,-0.0011608803179115057,0.0339018851518631,0.00656293286010623,-0.029130009934306145,-0.01834671013057232,-0.023907840251922607,0.036898914724588394,0.006084490101784468,-0.06746455281972885,0.03775486350059509,0.007233446929603815,-0.00519143370911479,-0.017874492332339287,0.002729092724621296,0.0026350365951657295,-0.015133967623114586,0.024510951712727547,0.03671843186020851,0.014198382385075092,0.018357431516051292,-0.03179648518562317,0.002592887496575713,0.023544922471046448,0.03399772197008133,-0.009678501635789871,-0.008132237009704113,0.03193535655736923,0.05600046366453171,0.002701335120946169,-0.03148706629872322,0.046580251306295395,0.016812250018119812,-0.024389510974287987,0.01296623982489109,-0.006053412798792124,-0.04332929849624634,0.05412972718477249,0.02130654826760292,-0.051173895597457886,-0.024204213172197342,-0.0020660115405917168,-0.013856770470738411,-0.007985253818333149,0.006529581733047962,0.04113410785794258,0.02723737061023712,0.0471951924264431,0.009231412783265114,-0.005537690129131079,0.01848517544567585,0.029482727870345116,-0.031166892498731613,0.047913916409015656,-0.015441862866282463,-0.03504522144794464,0.04951288551092148,-0.02016097493469715,0.04208211973309517,0.03168774023652077,-0.02055872045457363,0.025973081588745117,0.027437657117843628,-0.03907141461968422,-0.026856202632188797,0.03728991374373436,0.03408259525895119,-0.002878576749935746,0.011486572213470936,0.05589548870921135,-0.03703826293349266,0.03446761891245842,-0.03146755322813988,-0.008050931617617607,0.022368982434272766,0.03449753671884537,0.032263126224279404,-0.06182189658284187,0.000973690184764564,-0.0006614601006731391,0.030941292643547058,0.007848957553505898,0.01975150965154171,0.0029160522390156984,-0.011966681107878685,-0.012918737716972828,0.06333692371845245,0.030834633857011795,-0.014380715787410736,-0.0067038205452263355,-0.012590644881129265,-0.023923179134726524,-0.0058224014937877655,-0.029720790684223175,-0.019567467272281647,0.01309096347540617,0.00909629836678505,0.005699025001376867,0.011251295916736126,-0.00981457345187664,-0.03313235566020012,-0.0078077721409499645,-0.07204210758209229,-0.0029118929523974657,-0.035454437136650085,0.05287546664476395,0.007763907313346863,0.007644020486623049,0.05749429762363434,-0.029187986627221107,-0.012735787779092789,0.03811585530638695,0.03527792915701866,-0.027659907937049866,0.005347908940166235,0.011880689300596714,0.025060106068849564,-0.042014092206954956,-0.00336996978148818,-0.04214657470583916,-0.020864007994532585,-0.009262527339160442,-0.027721384540200233,0.005225831177085638,0.017071779817342758,0.002549303462728858,-0.0641605332493782,-0.035461194813251495,0.04008122533559799,-0.004937315359711647,0.002543143928050995,-0.013738568872213364,0.015776868909597397,0.01552758738398552,-0.04678342491388321,-0.012680387124419212,-0.028650250285863876,0.03254462406039238,-0.05433327332139015,-0.005706819240003824,-0.00490932771936059,0.010172586888074875,0.061442047357559204,0.033262819051742554,-0.014533093199133873,-0.006028154864907265,0.004096025135368109,0.03620338812470436,-0.023091493174433708,0.008847813121974468,0.009468479081988335,0.007319691590964794,0.050240565091371536,0.009634903632104397,-0.042890775948762894,-0.012817637994885445,0.0022374612744897604,0.009537805803120136,-0.033264659345149994,-0.011530455201864243,-0.05217624455690384,-0.03946400433778763,0.01215045340359211,0.024777112528681755,0.050570689141750336,-0.02897421084344387,-0.050906017422676086,-0.014214009046554565,0.04693643003702164,0.06142549216747284,0.009318794123828411,0.015928929671645164,-0.048806704580783844,0.04464345425367355,-0.02569648064672947,0.046130165457725525,-0.0015152804553508759,0.008579264394938946,-0.04096235707402229,-0.0028545353561639786,0.06542423367500305,-0.021412717178463936,-0.02040879987180233,0.054883286356925964,-0.000925844011362642,0.015060977078974247,0.038437265902757645,-0.028782211244106293,-0.05493182688951492,-0.017014119774103165,0.00009713972394820303,0.03452686592936516,0.007872148416936398,0.002899136859923601,0.026452815160155296,0.00020451376622077078,-0.026918379589915276,-0.006386998575180769,-0.0014318604953587055,-0.06078777462244034,-0.023632893338799477,-0.02729949727654457,0.02967141382396221,0.043731171637773514,-0.01861611008644104,-0.046601034700870514,-0.04715033993124962,-0.015617847442626953,0.07671661674976349,0.014872304163873196,0.04095400869846344,-0.012043018825352192,0.059269096702337265,0.023487389087677002,0.006868474185466766,-0.017504563555121422,-0.03867576643824577,-0.067263662815094,-0.040846116840839386,0.03411763533949852,0.005163436755537987,0.0098473085090518,0.004972629249095917,0.01867998205125332,0.03997723385691643,0.003269788809120655,-0.04365710914134979,0.06149668991565704,-0.021164409816265106,-0.04845931753516197,-0.019964292645454407,-0.01993476413190365,-0.019803361967206,-0.030084650963544846,0.03329579159617424,0.02241353876888752,0.03419812023639679,0.009252945892512798,-0.0592304952442646,-0.05189787596464157,0.0071650356985628605,-0.002765495330095291,-0.05365932360291481,0.03608134388923645,-0.04374563694000244,-0.014609728008508682,0.026538440957665443,0.001699291286058724,-0.03529850021004677,0.04011330008506775,-0.07265469431877136,-0.005695386789739132,0.02637159265577793,-0.012696919031441212,-0.02917470596730709,-0.0491512268781662,0.02204550988972187,-0.033525120466947556,-0.08609600365161896,-0.00005253954805084504,0.025533607229590416,-0.015115694142878056,-0.031634651124477386,-0.04481995850801468,0.05966715142130852,0.014592954888939857,-0.001813091803342104,-0.024922693148255348,0.007462723180651665,-0.015543274581432343,0.032095856964588165,-0.02544519677758217,-0.013318919576704502,0.00004619492756319232,-0.03148382529616356,-0.07066292315721512,0.04971718043088913,0.019511308521032333,-0.034731753170490265,-0.013545912690460682,-0.020269928500056267,0.04194397106766701,0.01997954025864601,0.023441132158041,0.03385905548930168,0.005472524557262659,0.012938491068780422,0.013428697362542152,-0.012136125937104225,0.020917614921927452,0.021176500245928764,0.00751491030678153,-0.04662105813622475,0.009862825274467468,0.027646396309137344,-0.03626888245344162,-0.002288372488692403,-0.009127592667937279,-0.014107993803918362,-0.017355920746922493,0.01918783411383629,0.034308746457099915,-0.015880601480603218,-0.035313479602336884,0.0022778187412768602,-0.022830864414572716,0.0039940280839800835,0.028721056878566742,-0.00858384184539318,0.0007414679857902229,-0.01847679540514946,0.07012959569692612,0.060135286301374435,0.017577165737748146,0.02210560441017151,-0.031280677765607834,-0.00973040796816349,-0.013429157435894012,-0.03800177946686745,-0.010603860020637512,-0.02385547012090683,0.029897764325141907,0.019158152863383293,0.023488804697990417,-0.05542445927858353,-0.0669543668627739,0.0023748925887048244,0.007516649551689625,-0.03698718547821045,0.037909574806690216,-0.06656228750944138,-0.038296788930892944,-0.021835077553987503,0.02068987675011158,-0.03479485586285591,-0.010884817689657211,0.057586923241615295,0.017257383093237877,-0.07652856409549713,0.014346186071634293,0.029030153527855873,0.03960766643285751,-0.03677678108215332,0.028362726792693138,0.0029402116779237986,0.021114487200975418,-0.030184712260961533,0.010771600529551506,0.013920847326517105,-0.057877685874700546,0.0032515625935047865,-0.03440778702497482,-0.0032484123948961496,-0.06087344139814377,-0.025744585320353508,-0.010577481240034103,-0.022510318085551262,-0.0205024816095829,0.014179199934005737,0.02037794515490532,0.03162766620516777,-0.017486846074461937,-0.021506600081920624,-0.04346197843551636,0.013681984506547451,-0.007041655480861664,0.019871579483151436,-0.019762173295021057,-0.019322363659739494,0.012797043658792973,0.016026431694626808,0.0024828400928527117,0.001994598191231489,-0.0000935438001761213,0.001235481700859964,-0.019079072400927544,-0.010357191786170006,0.014774340204894543,0.004453778732568026,-0.033695630729198456,0.023320725187659264,0.00673016207292676,0.003430004697293043,-0.008156470023095608,0.046684153378009796,-0.013029730878770351,-0.007383184973150492,0.05063381791114807,0.031144188717007637,0.013749145902693272,0.03521595522761345,0.03425253927707672,-0.036098260432481766,-0.01324842032045126,-0.012976503930985928,0.001436883700080216,0.042038701474666595,-0.016502998769283295,0.023145845159888268,0.053280819207429886,0.004892989527434111,0.04204675555229187,0.01346671860665083,-0.056764665991067886,0.013469935394823551,0.04839387536048889,-0.035006288439035416,0.025391673669219017,-0.02184494212269783,0.013700507581233978,-0.024083023890852928,0.011163738556206226,-0.007015731651335955,0.007868098095059395,0.017517948523163795,0.03205917030572891,-0.02753651700913906,-0.003307666163891554,0.01585829257965088,0.023229064419865608,0.04762186110019684,0.007560855709016323,-0.026082046329975128,-0.044081881642341614,0.03151514008641243,0.005582604557275772,-0.02597399428486824,-0.028118573129177094,-0.020766645669937134,0.023620661348104477,0.004692520014941692,0.020319990813732147,0.0209431741386652,-0.020681090652942657,0.035810064524412155,-0.025014603510499,-0.02713141217827797,-0.01138665433973074,0.012744291685521603,-0.04114660620689392,0.0555449016392231,0.01619747467339039,0.05006308853626251,0.03895962983369827,-0.015140826813876629,0.00795693602412939,-0.0308376532047987,0.020626230165362358,0.021412495523691177,0.008729408495128155,-0.03327800706028938,-0.018907511606812477,0.020541131496429443,0.0075831059366464615,0.04074893146753311,0.011486807838082314,0.02470875345170498,0.05767052620649338,-0.0019741966389119625,-0.03789506480097771,0.041246768087148666,0.012363409623503685,-0.0013948489213362336,0.0790334939956665,-0.012265403755009174,0.006180629134178162,0.001153795514255762,0.014195720665156841,0.020706091076135635,0.01892125979065895,-0.01232197880744934,0.04838307574391365,-0.006491804961115122,-0.0018355125794187188,0.02000386454164982,-0.017548179253935814,-0.02514447271823883,0.05452992767095566,0.03966442123055458,-0.009572414681315422,0.05687291547656059,-0.014746692962944508,0.01669693924486637,0.035272371023893356,0.04165846109390259,0.00854187086224556,0.006210825406014919,0.030956147238612175,-0.012956508435308933,-0.05160393938422203,-0.06434262543916702,0.008920329622924328,-0.03360747545957565,0.058390967547893524,-0.006592695135623217,0.026881558820605278,-0.057151149958372116,0.029284309595823288,0.006882982794195414,0.019870825111865997,-0.012273830361664295,-0.039823129773139954,-0.021828044205904007,0.050154298543930054,0.04382377490401268,0.017346849665045738,-0.031824741512537,-0.015133224427700043,-0.03666399419307709,0.0037356107495725155,0.005541981663554907,-0.0007753863465040922,0.012753917835652828,0.01717524416744709,0.00770985521376133,0.04131108894944191,-0.0007042647921480238,0.0009259225334972143,-0.026984145864844322,-0.016777828335762024,0.02703642100095749,-0.007557446137070656,-0.04268626496195793,0.029617460444569588,0.07847153395414352,-0.02301955036818981,-0.04188394173979759,-0.008323313668370247,0.03810012340545654,0.0007715989486314356,0.017002252861857414,-0.0026289692614227533,-0.06788001209497452,-0.038502179086208344,-0.048668839037418365,-0.0015885427128523588,-0.0033985150512307882,-0.022613491863012314,-0.013083158992230892,-0.05197376012802124,-0.004135928116738796,0.012109478004276752,-0.0015456427354365587,0.008539789356291294,0.0002677702868822962,-0.020971648395061493,0.012707434594631195,0.011308379471302032,0.025012921541929245,0.011372009292244911,-0.019175846129655838,-0.012453141622245312,-0.055596500635147095,0.00884331576526165,0.006224530749022961,-0.017740588635206223,-0.017487341538071632,0.0015366218285635114,0.03870658949017525,-0.04353008791804314,0.002622681437060237,-0.00989013072103262,0.013730771839618683,-0.002127062762156129,-0.03387915715575218,0.023337552323937416,-0.00996981281787157,-0.001484190346673131,-0.05960904061794281,0.008270039223134518,0.023740720003843307,-0.00661700451746583,-0.03235800191760063,0.03796816244721413,-0.006449640262871981,-0.011636799201369286,0.02602686546742916,0.01581929251551628,-0.024983417242765427,-0.009117105975747108,0.020950933918356895,-0.029842685908079147,0.006392786279320717,-0.005244828294962645,-0.02831498347222805,-0.02325107529759407,0.05245857313275337,-0.006914802826941013,0.018559297546744347,-0.00040502738556824625,-0.04034990444779396,0.002448797458782792,-0.0018274489557370543,0.004474060144275427,0.05412135273218155,0.07130251824855804,-0.047640472650527954,-0.03718689829111099,-0.043199919164180756,-0.004179026931524277,0.009737487882375717,0.037441495805978775,-0.03308514878153801,-0.008956972509622574,-0.05989664047956467,-0.058420006185770035,-0.03716837614774704,0.002483043121173978,0.021411806344985962,-0.026307091116905212,0.029177820309996605,-0.03980238735675812,0.00595121318474412,0.004190587438642979,0.026525918394327164,0.06296700984239578,-0.002296525752171874,0.02707265317440033,0.05312168598175049,0.0005989731871522963,-0.007343186065554619,0.009389160200953484,0.022762104868888855,0.01421570684760809,-0.011257171630859375,0.028546271845698357,-0.04071471095085144,-0.061915718019008636,-0.021429026499390602,-0.01055402122437954,-0.06735667586326599,0.015046488493680954,-0.021838447079062462,-0.007646446581929922,-0.002060451079159975,0.0028171103913336992,0.01929500699043274,-0.003464998910203576,0.005015803035348654,0.025509465485811234,-0.012002247385680676,-0.009782264940440655,-0.055431026965379715,0.008878294378519058,0.033467311412096024,-0.03847191855311394,-0.009559128433465958,-0.009282493963837624,0.017602382227778435,-0.0266120545566082,0.0482129231095314,-0.004363623913377523,0.003095246385782957,-0.0819537565112114,0.009795584715902805,-0.026535633951425552,0.02175360545516014,-0.005084949545562267,0.009896190837025642,-0.00937906838953495,0.011558756232261658,0.029789738357067108,0.034542832523584366,0.0026812751311808825,0.03981989622116089,0.008087879046797752,-0.004094259813427925,0.0007567332941107452,-0.0035219008568674326,-0.07854929566383362,0.04008615016937256,-0.02664117142558098,0.05538983643054962,0.0489402711391449,0.06403981149196625,-0.017244113609194756,0.024771438911557198,-0.003933339845389128,-0.08582651615142822,-0.022719332948327065,0.01640693098306656,-0.0675729513168335,-0.01741882972419262,0.028382079675793648,-0.03020656481385231,0.008066962473094463,-0.017684293910861015,0.010968138463795185,-0.010424382984638214,0.033096399158239365,-0.02716125175356865,0.004180428571999073,0.016327712684869766,0.02270348370075226,-0.04268091917037964,-0.013400568626821041,-0.027883833274245262,0.03140544146299362,0.008515234105288982,-0.0253548976033926,-0.0024636215530335903,-0.008501183241605759,-0.0060722436755895615,0.012958422303199768,0.015581125393509865,-0.014780372381210327,-0.061727214604616165,-0.001506305648945272,0.016809390857815742,0.01941763609647751]

**Comment by zouhuigang** [comment]
使用filter去查询数据时，有时候搜索不出来，把Filter条件去掉之后，再重新搜索可以出来

**Reply by yhmo** [reply]
你把向量字段的索引换成 FLAT 就能搜出来了。
**划重点：用IVF/HNSW索引做过滤查询，不能保证一定能搜出全部结果，尤其是当符合条件的数据条数很少时。**

原理：
不管是IVF还是HNSW，搜索的时候都是从“跟目标向量接近的地方开始找”，它们只搜索一个小范围，避免全量的计算，从而获得10倍以上的性能提升。但是，符合过滤条件的数据未必是在“跟目标向量接近的地方“，而不管是IVF还是HNSW，它们搜索的范围都是有限的，所以就有可能找不到符合过滤条件的数据。

如果你想找到所有符合过滤条件的数据，只能使用FLAT，或者在IVF索引里把nprobe=nlist
